### DISCUSSION TITLE
Top Navigation - horizontal line separator within a grouped menu

### DISCUSSION THREAD

**Discussion by Fzoltan87** [filamentphp/filament#13018] [Help]
### Package

Panel builder

### Package Version

3.2.84.

### How can we help you?

I have a grouped menu item called "Landing Page".
How can I solve it so that I can separate the menu items with a horizontal line within the "Landing Page" menu item.

This is where I implement the menu bar: app\Providers\Filament\AdminPanelProvider.php

```
->navigation(function (NavigationBuilder $navigationBuilder) {
    return $navigationBuilder
        ->items([
            ...Pages\Dashboard::getNavigationItems(),
        ])
        ->groups([
            NavigationGroup::make('Landing Page')
                ->items([
                    ...($user?->can('read frontend_contact') ? ContactResource::getNavigationItems() : []),
                    ...($user?->can('read frontend_content') ? ContentResource::getNavigationItems() : []),
                    ...($user?->can('read frontend_page') ? PageResource::getNavigationItems() : []),
                    --------------------------- horizontal line separator here
                    ...($user?->can('read blog_category') ? CategoryResource::getNavigationItems() : []),
                    ...($user?->can('read blog_post') ? PostResource::getNavigationItems() : []),
            ]),
        ]),
```

I want to implement the structure:
Landing Page
  - Page
  - Content
  - Contact
  --------------  <- horizontal line separator
  - Category
  - Blog

**Comment by simbabanda** [comment]
As far as I can tell, you can only add html for groups, not items... If you'd like to do it for groups, here's the code below:

`NavigationGroup::make()
 ->label('Landing Page')
 ->extraSidebarAttributes(['class' => 'nav-group']),`

In your theme.css or whatever you call your main css file, then have the class defined something like this:
```
.nav-group::before {
    content: "";
    @apply block bg-baxtr-blue mb-1 h-[0.005em] opacity-25 w-full;
}
```

I hope that helps

**Reply by Fzoltan87** [reply]
Hallo.

This puts the frame on the menu.

I should place it within the list of sub-menu items.

```
NavigationGroup::make('Landing Page')
->items([
...($user?->can('read frontend_contact') ? ContactResource::getNavigationItems() : []),
])->extraTopbarAttributes(['class' => 'nav-group']),
```

![Screenshot 2024-07-25 at 14-31-06 Dashboard - Your Brand](https://github.com/user-attachments/assets/8c311a7d-d7ab-4b66-829e-1b3fc84f1568)

**Reply by Joao-V-Queiroz** [reply]
Hello, 

Filament resolved this in these new versions, here is an example:

            ActionGroup::make([
                Actions\Action::make('Cancelar Perícia')
                    ->label('Cancelar Perícia')
                    ->icon('heroicon-o-exclamation-triangle')
                    ->iconSize('sm')
                    ->color('warning')
                    ->visible(fn(Pericia $record) => $record->status === 'agendada' || $record->status === 'aguardando junta medica')
                    ->form([
                        Textarea::make('desc_cancel')
                            ->label('Motivo do cancelamento')
                            ->rows(2)
                            ->maxLength(255)
                            ->required(),
                    ])
                    ->action(function (array $data, Pericia $record) {
                        $record->update(['status' => 'cancelada']);
                        $record->desc_cancel = $data['desc_cancel'];
                        $record->save();
                        Notification::make()
                            ->title('Perícia cancelada')
                            ->success()
                            ->send();
                    }),
                Actions\Action::make('faltou')
                    ->label('Faltou')
                    ->icon('heroicon-o-x-circle')
                    ->iconSize('sm')
                    ->color('danger')
                    ->visible(fn(Pericia $record) => $record->status === 'agendada' || $record->status === 'aguardando junta medica')
                    ->requiresConfirmation()
                    ->modalHeading('Registrar Falta')
                    ->modalDescription('Tem certeza que deseja registrar falta do interessado?')
                    ->modalSubmitActionLabel('Sim, registrar falta')
                    ->action(function (Pericia $record) {
                        $record->update(['status' => 'faltou']);
                        Notification::make()
                            ->title('Falta registrada')
                            ->success()
                            ->send();
                    }),
                ActionGroup::make([
                    Actions\DeleteAction::make(),
                ])->dropdown(false),
            ])
            
 
![image](https://github.com/user-attachments/assets/8f8e1e1a-36b0-4dfa-b58b-4fa1dcd66f6e)
